Adding `license_expression` and `license_files` serialization

idavis opened the pull request:
Adds `license_expression` and `license_files` to the pyproject matadata specified in [PEP 639 Core Metadata](https://peps.python.org/pep-0639/#id1). No validation is performed as that should be handled by consumers. Also adds default value for `license_files` as described in PEP 639.

Though the `paths` and `globs` are mutually exclusive through validation, serialization allows them both to be specified. I could not find a way to restrict this via the toml-rs serializer. I think there is a bug in the dotted key handling.

messense commented:
> Though the paths and globs are mutually exclusive through validation, serialization allows them both to be specified. 

Maybe we should use an enum instead?

idavis commented:
@messense I tried to use an enum but the dotted names caused issues with the serialization.

messense commented:
We can also migrate to toml_edit: https://github.com/ordian/toml_edit

We already done that in maturin.

idavis commented:
@messense So you are suggesting that we change the `maturin/src/pyproject_toml.rs` to use `toml_edit` instead of this repo?

idavis commented:
@messense I have converted the repo to use `toml_edit` just like `maturin`. I was then able to convert the licenses metadata to an enum and added tests.

messense commented:
Thanks! Since PEP 639 is still a draft [waiting for review](https://discuss.python.org/t/pep-639-round-2-improving-license-clarity-with-better-package-metadata/12622/61), let's use git dependency in maturin for implementing multiple license files support for now.
